omnigibson/prims/entity_prim.py

```diff
@@ -955,76 +956,117 @@ def get_relative_linear_velocity(self):
+    ):
+        """
+        Set the position and orientation of entry prim object.
+
+        Args:
+            position (None or 3-array): The position to set the object to. If None, the position is not changed.
+            orientation (None or 4-array): The orientation to set the object to. If None, the orientation is not changed.
+            frame (Literal): The frame in which to set the position and orientation. Defaults to world. parent frame
+            set position relative to the object parent. scene frame set position relative to the scene.
+        """
+
+        # If we are in a scene, compute the scene-local transform before setting the pose
+        if frame == "scene" and self.scene is not None:
+
+            # if position or orientation is None, get the current position and orientation relative to scene
+            current_position, current_orientation = self.get_position_orientation(frame="scene")
+            position = current_position if position is None else np.array(position, dtype=float)
+            orientation = current_orientation if orientation is None else np.array(orientation, dtype=float)
+
+            position, orientation = T.pose_transform(*self.scene.prim.get_position_orientation(), position, orientation)
 
-    def set_position_orientation(self, position=None, orientation=None):
         # If kinematic only, clear cache for the root link
         if self.kinematic_only:
             self.root_link.clear_kinematic_only_cache()
         # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
         if og.sim.is_stopped():
-            XFormPrim.set_position_orientation(self, position=position, orientation=orientation)
+            return XFormPrim.set_position_orientation(self, position=position, orientation=orientation, frame=frame)
         # Delegate to RigidPrim if we are not articulated
         elif self._articulation_view is None:
-            self.root_link.set_position_orientation(position=position, orientation=orientation)
+            self.root_link.set_position_orientation(position=position, orientation=orientation, frame=frame)
         # Sim is running and articulation view exists, so use that physx API backend
         else:
             if position is not None:
                 position = np.asarray(position)[None, :]
             if orientation is not None:
                 orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
-            self._articulation_view.set_world_poses(position, orientation)
+
+            if frame == "world" or frame == "scene":
+                self._articulation_view.set_world_poses(position, orientation)
+            else:
+                self._articulation_view.set_local_poses(position, orientation)
+
             PoseAPI.invalidate()
 
-    def get_position_orientation(self):
+    def get_position_orientation(self, frame: Literal["world", "scene", "parent"] = "world"):
```

Comment:
I think "Literal" doesn't actually enforce the frame to be one of the three.

You should add explicit checking in your code.